src/sdk/namespace_fs.js

```diff
@@ -459,11 +570,27 @@ class NamespaceFS {
 
                 if (cached_dir.sorted_entries) {
                     const sorted_entries = cached_dir.sorted_entries;
-                    const marker_index = _.sortedLastIndexBy(
-                        sorted_entries,
-                        make_named_dirent(marker_curr),
-                        get_entry_name
-                    );
+                    let marker_index;
+                    // Two ways followed here to find the index.
+                    // 1. When inside marker_dir: Here the entries are sorted based on time. Here
+                    //    FindIndex() is called since sortedLastIndexBy() expects sorted order by name
+                    // 2. When marker_dir above dir_path: sortedLastIndexBy() is called since entries are
+                    //     sorted by name
+                    if (list_versions && marker_curr && !marker_curr.includes('/')) {
```

Comment:
shouldn't we use _.findIndex() only when version_id_marker passed?

Reply:
When only `key_marker` is provided then we got to list the versions of all the versions of that `key_marker`. So in this case, assigning `start_marker` to `marker_curr`. If `version_id_marker` is also provided then assigning `version_id_marker` to `marker_curr`

Correct me if wrong

Reply:
got you, right 
and can you please add a comment about this case - `&& !marker_curr.includes('/')` 
not sure I understand it

Reply:
This will make sure that the path does not include `/`. 

ex:  if `key_marker = dir1/file1`, we call `findIndex()` only when we get at file level i.e when `marker_curr = file1`. `marker_curr` is directory by directory search and traverses through `key_marker` path. We dont call findIndex() when `marker_curr = dir1/file1` instead call goest to else part.

Reply:
ok, can you add a comment on that case?